Scripts/Runtime/Core/LongGuid.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using UnityEngine;
+
+namespace BrunoMikoski.ScriptableObjectCollections
+{
+    [Serializable]
+    public struct LongGuid : IEquatable<LongGuid>
+    {
+        [SerializeField]
+        private long value1;
+        [SerializeField]
+        private long value2;
+
+        public LongGuid(long guidValue1, long guidValue2)
+        {
+            value1 = guidValue1;
+            value2 = guidValue2;
+        }
+
+        public static LongGuid NewGuid()
+        {
+            Guid guid = Guid.NewGuid();
+            byte[] guidBytes = guid.ToByteArray();
+            long guidValue1 = BitConverter.ToInt64(guidBytes, 0);
+            long guidValue2 = BitConverter.ToInt64(guidBytes, 8);
+
+            return new LongGuid(guidValue1, guidValue2);
+        }
+
+        public (long, long) GetValue()
+        {
+            return (value1, value2);
+        }
+
+        public bool IsValid()
+        {
+            return value1 != 0 || value2 != 0;
```

Comment:
Should only one of these values be non-zero or both?

Reply:
Since the objective for this is to be super performatic, checking first for the initial and then for the second one is fine, and since those are two private variables that are only written here, it should be safe.